inst/cpp/sim.cpp

```diff
@@ -84,11 +84,14 @@ List sim_wrapper_cpp (NumericVector A, List design, List par, NumericVector iov_
   prv_dose = doses[0];
   t_prv_dose = times[0];
 
-  // Main call to ODE solver
-  ode(A_dum, dAdt_dum, 0);
-
   // insert_state_init
   NumericVector Aupd = clone(A);
+  
```

Comment:
this part is just re-ordering. We first need to initialize the state, and then call the ode solver once to initialize the derived variables. It was the opposite way. I don't see a reason why this change could lead to issues since there is no reasonable way the initial state should depend on something in the ODE block, it can only be the other way around.